### PR TITLE
MOM: Backup keyword and cards

### LINKED ISSUE #2774: MOM: Backup Mechanic

### PR DESCRIPTION
**Pull request by Northmoc**
- closes #2774

### CODE REVIEW

File: forge-game/src/main/java/forge/game/keyword/Keyword.java [hunk 1]
```diff
@@ -28,6 +28,7 @@ public enum Keyword {
     ASSIST("Assist", SimpleKeyword.class, true, "Another player can pay up to %s of this spell's cost."),
     AURA_SWAP("Aura swap", KeywordWithCost.class, false, "%s: You may exchange this Aura with an Aura card in your hand."),
     AWAKEN("Awaken", KeywordWithCostAndAmount.class, false, "If you cast this spell for %s, also put {%d:+1/+1 counter} on target land you control and it becomes a 0/0 Elemental creature with haste. It's still a land."),
+    BACKUP("Backup", KeywordWithAmountAndType.class, false, "When this creature enters the battlefield, put {%1$d:+1/+1 counter} on target creature. If that's another creature, it gains the following ability until end of turn."),
```
Comment (Hanmac) [inline forge-game/src/main/java/forge/game/keyword/Keyword.java:31]: shouldn't it be just withAmount or does it mess up the description?

Comment (Northmoc) [reply forge-game/src/main/java/forge/game/keyword/Keyword.java:31]: Crashes with `KeywordWithAmount` and Backup:2:**THIS**

Comment (Hanmac) [reply forge-game/src/main/java/forge/game/keyword/Keyword.java:31]: Hmm I don't know If that's okay for now

I would prefer the Type ones only for the ones with types

@tehdiplomat your opinion?

File: forge-gui/res/cardsfolder/upcoming/archpriest_of_shadows.txt [hunk 2]
```diff
@@ -0,0 +1,13 @@
+Name:Archpriest of Shadows
+ManaCost:3 B B
+Types:Creature Human Warlock
+PT:4/4
+K:Backup:BackupAbilities
```
Comment (tehdiplomat) [inline forge-gui/res/cardsfolder/upcoming/archpriest_of_shadows.txt:5]: Missing the number here

File: forge-game/src/main/java/forge/game/keyword/Keyword.java (hunk 1 above)
Comment (tehdiplomat) [reply forge-game/src/main/java/forge/game/keyword/Keyword.java:31]: Yea I feel like we'd need an extra Keyword class for this maybe `KeywordWithAmountAndOptions` or something like that, where we can just pass "extra things" that might get handled a bit different depending on the keyword.

Comment (Northmoc) [reply forge-game/src/main/java/forge/game/keyword/Keyword.java]: There was already some support for "Extra" on KeywordWithAmount, so I expanded that

File: forge-gui/res/cardsfolder/upcoming/archpriest_of_shadows.txt [hunk 3]
```diff
@@ -0,0 +1,14 @@
+Name:Archpriest of Shadows
+ManaCost:3 B B
+Types:Creature Human Warlock
+PT:4/4
+K:Backup:2:BackupAbilities
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/archpriest_of_shadows.txt:5]: should still be 1?

**Comment by Hanmac** [comment]
For `PlayMain1`: @Agetian should try to add AI logic that Ai should only play a Backup card as Main 1 if it has creatures that could attack?

**Comment by Northmoc** [comment]
@Agetian I took a try at implementing the logic Hanmac suggested... see what you think 😄

**Comment by Agetian** [comment]
Yep, should work nice :) :+1:

**Review by Agetian — APPROVED** [review]

**Pull request merged**